File: DevExpress-Demo/VB/ChartsDemo.Wpf/Modules/ChartElements/LogarithmicScaleViewModel.vb
Imports System.Windows.Media
Imports DevExpress.Mvvm.POCO
Imports DevExpress.Xpf.Charts

Namespace ChartsDemo

    Public Class LogarithmicScaleViewModel

        Public Shared Function Create() As LogarithmicScaleViewModel
            Return ViewModelSource.Create(Function() New LogarithmicScaleViewModel())
        End Function

        Public Overridable Property Palette As Palette

        Public Overridable Property Headphones190dBSPLBrush As SolidColorBrush

        Public Overridable Property Headphones1100dBSPLBrush As SolidColorBrush

        Public Overridable Property Headphones290dBSPLBrush As SolidColorBrush

        Public Overridable Property Headphones2100dBSPLBrush As SolidColorBrush

        Protected Sub New()
            Palette = New OfficePalette()
        End Sub

        Protected Sub OnPaletteChanged()
            If Palette Is Nothing Then Return
            Headphones190dBSPLBrush = New SolidColorBrush(Palette(0))
            Headphones1100dBSPLBrush = New SolidColorBrush(ChangeColorLuminance(Palette(0), 0.8F))
            Headphones290dBSPLBrush = New SolidColorBrush(Palette(1))
            Headphones2100dBSPLBrush = New SolidColorBrush(ChangeColorLuminance(Palette(1), 0.8F))
            Headphones190dBSPLBrush.Freeze()
            Headphones1100dBSPLBrush.Freeze()
            Headphones290dBSPLBrush.Freeze()
            Headphones2100dBSPLBrush.Freeze()
        End Sub
    End Class
End Namespace

File: DevExpress-Demo/CS/ChartsDemo.Wpf/Modules/ChartElements/LogarithmicScaleViewModel.cs
using System.Windows.Media;
using DevExpress.Mvvm.POCO;
using DevExpress.Xpf.Charts;

namespace ChartsDemo {
    public class LogarithmicScaleViewModel {
        public static LogarithmicScaleViewModel Create() {
            return ViewModelSource.Create(() => new LogarithmicScaleViewModel());
        }

        public virtual Palette Palette {
            get;
            set;
        }
        public virtual SolidColorBrush Headphones190dBSPLBrush {
            get;
            protected set;
        }
        public virtual SolidColorBrush Headphones1100dBSPLBrush {
            get;
            protected set;
        }
        public virtual SolidColorBrush Headphones290dBSPLBrush {
            get;
            protected set;
        }
        public virtual SolidColorBrush Headphones2100dBSPLBrush {
            get;
            protected set;
        }

        protected LogarithmicScaleViewModel() {
            Palette = new OfficePalette();
        }

        protected void OnPaletteChanged() {
            if (Palette == null)
                return;
            Headphones190dBSPLBrush = new SolidColorBrush(Palette[0]);
            Headphones1100dBSPLBrush = new SolidColorBrush(ColorUtils.ChangeColorLuminance(Palette[0], 0.8f));
            Headphones290dBSPLBrush = new SolidColorBrush(Palette[1]);
            Headphones2100dBSPLBrush = new SolidColorBrush(ColorUtils.ChangeColorLuminance(Palette[1], 0.8f));
            Headphones190dBSPLBrush.Freeze();
            Headphones1100dBSPLBrush.Freeze();
            Headphones290dBSPLBrush.Freeze();
            Headphones2100dBSPLBrush.Freeze();
        }
    }
}
